tests/check-docs.rs

```diff
@@ -0,0 +1,60 @@
+use regex::Regex;
+use std::{collections::BTreeMap, fmt::Write, path::PathBuf};
+use walkdir::WalkDir;
+
+fn mangle_comments(s: &str) -> String {
+    let result = s
+        .split('\n')
+        .map(str::trim)
+        .filter_map(|line| line.strip_prefix("/// "))
+        .filter(|line| !line.starts_with("# ")) // filter out code
+        .map(|line| if line.starts_with("```") { "" } else { line })
```

Comment:
I think it would be cool to add indention to code blocks like this:

```
### Example

    let mut v = Vec::new();
    v.push(0);

Use instead:

    let v = vec![0];
```

Reply:
I chose to leave the `\`\`\`` for now. `rustc --explain` does the same, so there is some precedent, and for this first version it should be OK.